psyneulink/core/globals/parameters.py

```diff
@@ -2124,9 +2124,18 @@ def __init__(self, owner, parent=None):
         for alias_name in aliases_to_create:
             setattr(self, alias_name, ParameterAlias(name=alias_name, source=getattr(self, alias_name).source))
 
-        for param, value in self.values(show_all=True).items():
+        values = self.values(show_all=True)
+        for param, value in values.items():
             self._validate(param, value.default_value)
 
+            if value.dependencies is not None:
```

Comment:
just a drive-by comment:
Can this be simplified?
```
value.dependencies = {getattr(d, 'name', d) for d in value.dependencies}
assert all(dep in values for dep in value.dependencies)
```

Reply:
Yes, thanks. I always forget about the default option for getattr.